buf/internal/toolchain.bzl

```diff
@@ -16,122 +16,83 @@
 
 load("@bazel_tools//tools/build_defs/repo:utils.bzl", "update_attrs")
 
-_TOOLCHAINS_REPO = "rules_buf_toolchains"
-
-_BUILD_FILE = """
-load(":toolchain.bzl", "declare_buf_toolchains")
+_TOOLS = ["buf", "protoc-gen-buf-breaking", "protoc-gen-buf-lint"]
 
-package(default_visibility = ["//visibility:public"])
+_TOOLCHAINS_REPO = "rules_buf_toolchains"
 
-declare_buf_toolchains(
-    os = "{os}",
-    cpu = "{cpu}",
-    rules_buf_repo_name = "{rules_buf_repo_name}",
- )
-"""
+BufToolchainInfo = provider(fields = {
+    "buf_tool": "A File for a buf executable",
+    "protoc_breaking_tool": "A File for protoc-gen-buf-breaking executable",
+    "protoc_lint_tool": "A File for protoc-gen-buf-lint executable",
+})
 
-_TOOLCHAIN_FILE = """
 def _buf_toolchain_impl(ctx):
-    toolchain_info = platform_common.ToolchainInfo(
-        cli = ctx.executable.cli,
+    return platform_common.ToolchainInfo(
+        buf = BufToolchainInfo(
```

Comment:
I think we can also include the old option to avoid breaking changes and mark it deprecated.

Reply:
i am not sure how this would be breaking, the previous api was not public